internal/consumer/service.go

```diff
@@ -0,0 +1,81 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/model"
+
+	"github.com/emmaLP/gs-software-onboarding/pkg/hackernews"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type Service struct {
+	logger          *zap.Logger
+	numberOfWorkers int
+	hnClient        hackernews.Client
+}
+
+type Client interface {
+	processStories(ctx context.Context)
+}
+
+func NewService(logger *zap.Logger, config *model.ConsumerConfig, hnClient hackernews.Client) (*Service, error) {
+	if hnClient == nil {
+		var err error
+		hnClient, err = hackernews.New(config.BaseUrl, nil)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to create HackerNew client: %w", err)
+		}
+	}
+	return &Service{
+		logger:          logger,
+		numberOfWorkers: config.NumberOfWorkers,
+		hnClient:        hnClient,
+	}, nil
+}
+
+func (s *Service) processStories(ctx context.Context) error {
```

Comment:
`ctx` is passed in but not used. Usually whenever a message is being sent to or received from a channel, it's paired with a `<- ctx.Done()` check to ensure we can exit the program. (Couple examples [one](https://gobyexample.com/context), [two](https://github.com/gymshark/realm-migration-service/blob/main/internal/worker/worker.go#L42))

An example here is:

```
for _, id := range storyIds {
	topStoriesChan <- id
}
```

If nothing is reading from `topStoriesChan` then this will sit and block indefinitely. When the program exits, this will continue to block resulting in the program not gracefully terminating